lib/sqlalchemy/dialects/sqlite/base.py

```diff
@@ -2617,15 +2617,10 @@ def get_check_constraints(self, connection, table_name, schema=None, **kw):
             connection, table_name, schema=schema, **kw
         )
 
-        CHECK_PATTERN = r"(?:CONSTRAINT (.+) +)?" r"CHECK *\( *(.+) *\),? *"
+        CHECK_PATTERN = r"(?:CONSTRAINT (\w+) +)?" r"CHECK *\( *((.|\n)+?) *\)(, ?\n|\n) *"
```

Comment:
I don't think `\w` is enough to obtain the name of the constraint, since it can include any char, like `constraint "foo~bar" CHECK (a>0)`

Also the double string `" r"` can be removed.

Not sure if the additional capture groups are also needed, maybe they could use `(?:...)`